independent-projects/qute/debug/src/main/java/io/quarkus/qute/debug/agent/SourceTemplateRegistry.java

```diff
@@ -0,0 +1,94 @@
+package io.quarkus.qute.debug.agent;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.lsp4j.debug.Source;
+
+public class SourceTemplateRegistry {
+
+    private final Map<String, Source> templateIdToSource = new HashMap<>();
+
+    private final List<String> basePaths;
+    private final List<String> fileExtensions;
+
+    public SourceTemplateRegistry() {
+        this(List.of("src/main/resources/templates/"),
+                List.of(".qute", ".html", ".qute.html", ".yaml", ".qute.yaml", ".yml", ".qute.yml", ".txt", ".qute.txt", ".md",
+                        ".qute.md"));
+    }
+
+    public SourceTemplateRegistry(List<String> basePaths, List<String> fileExtensions) {
+        this.basePaths = basePaths;
```

Comment:
As Qute doesn't store the template file path, but only template id, I need to manage a mapping between template id and full file path):

 * basePaths will search templates from those basePaths
 * file extension 

It would be super nice if Qute could manage those file path.